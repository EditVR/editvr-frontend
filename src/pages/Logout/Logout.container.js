/**
 * @file Logout.container.js
 * Exports a redux-connected Logout component.
 */

import { connect } from 'react-redux';
import Logout from './Logout';

const mapDispatchToProps = dispatch => ({
  dispatch
});

const mapState = () => ({});

export default connect(mapState, mapDispatchToProps)(Logout);
