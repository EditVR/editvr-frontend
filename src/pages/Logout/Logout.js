/**
 * @file Logout.js
 * Exports a React component that handles requests to /logout.
 */

import { Component } from 'react';
import PropTypes from 'prop-types';

import { USER_LOG_OUT } from '../../constants';

class Logout extends Component {
  static propTypes = {
    dispatch: PropTypes.func.isRequired
  };

  /**
   * {@inheritdoc}
   */
  componentWillMount() {
    this.props.dispatch({ type: USER_LOG_OUT });
  }

  /**
   * {@inheritdoc}
   */
  render() {
    return null;
  }
}

export default Logout;
