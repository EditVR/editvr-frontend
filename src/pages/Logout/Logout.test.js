/**
 * @file Logout.test.js
 * Contains tests for Logout.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';

import Logout from './Logout.container';

describe('<Logout />', () => {
  it('Matches its snapshot', () => {
    const store = configureStore()({
      user: {
        uid: '1',
        authentication: {
          accessToken: 'test',
          csrfToken: 'test'
        }
      }
    });
    expect(
      renderer
        .create(
          <Provider store={store}>
            <Router>
              <Logout />
            </Router>
          </Provider>
        )
        .toJSON()
    ).toMatchSnapshot();
  });
});
