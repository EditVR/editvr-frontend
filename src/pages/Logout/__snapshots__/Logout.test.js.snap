// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logout /> Matches its snapshot 1`] = `null`;
